web_widget_boolean_switch/__openerp__.py

```diff
@@ -0,0 +1,177 @@
+# -*- coding: utf-8 -*-
+# © <YEAR(S)> <AUTHOR(S)>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+{
+    "name": "Web widget boolean switch",
+    "summary": "This module add widget to render boolean fields",
+    "version": "7.0.1.0.0",
+    "category": "web",
+    "website": "https://odoo-community.org/",
+    "author": "<pverkest@anybox>, Odoo Community Association (OCA)",
+    "license": "AGPL-3",
+    "application": False,
+    "installable": True,
+    "external_dependencies": {
+        "python": [],
+        "bin": [],
+    },
+    "depends": [
+        "base",
+        "web",
+    ],
+    "data": [
+    ],
+    "js": [
+        'static/lib/bootstrap-switch/bootstrap-switch.js',
+        'static/src/js/web_widget_boolean_switch.js',
+    ],
+    'test': [
+        'static/test/web_widget_boolean_switch.js',
+    ],
+    "css": [
+        'static/lib/bootstrap-switch/bootstrap-switch.css',
+    ],
+    'qweb': [
+        'static/src/xml/web_widget_boolean_switch.xml',
+    ],
+    "demo": [
+        'demo/res_users_view.xml',
+    ],
+    'description': """
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+=========================
+Web widget boolean switch
+=========================
+
+This module add a widget ``boolean_switch`` to render boolean fields. One
+of it's main features is to quick edit that field without enter in edit mode
+from list view or form view.
+
+
+Configuration
+=============
+
+In the view (test on tree view and form view), you can declare any boolean
+field using this widget.
+
+Example
+-------
+
+```
+   <field name="active"
+          widget="boolean_switch"
+          attrs="{'readonly': [('login', '=', 'admin')]}"
+          context="{'fake_parameter': 'foo'}"
+          options="{'quick_edit': True, 'extra': {'onText': 'Yes',
+                                                  'offText': 'No'}}"/>
+```
+
+Options
+-------
+
+quick_edit
+~~~~~~~~~~
+
+extra
+~~~~~
+``extra`` is used to set
+`bootstrap-switch <http://www.bootstrap-switch.org/options.html>`_ options.
+
+*Available options*:
+
+   * **size**: The checkbox size - default: `null` - values: null, 'mini',
+     'small', 'normal', 'large'
+   * **animate**: Animate the switch - default: `true`
+   * **indeterminate**: Indeterminate state - default: `false`
```

Comment:
I'm not sure this options has much sense for Odoo since it has no indeterminate values.